### PR TITLE
Service config as code docs

### PR DESCRIPTION
**Pull request by coffee-cup**
Add a separate page that documents how to use a `railway.toml` and `railway.json` file.

https://railway-docs-docs-pr-161.up.railway.app/deploy/config-as-code

Todo
- [x] Add syntax highlighting for toml and json code snippets

### CODE REVIEW

File: src/pages/deploy/config-as-code.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
+---
+title: Config as Code
+---
+
+<PriorityBoardingBanner />
+
+Railway supports defining the configuration for a single deployment in a file alongside your code. By default, we will look for a `railway.toml` or `railway.json` file. Everything in the build and deploy sections of the service settings page can be specified in this configuration file.
+
+_This is a beta feature and may change in the near future. Use at your own risk._
```
Comment (ndneighbor) [inline src/pages/deploy/config-as-code.md:9]: nit: Wouldn't the banner already suffice?

File: src/pages/deploy/config-as-code.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+## Examples
+
+These configuration files are equivalent.
```
Comment (ndneighbor) [inline src/pages/deploy/config-as-code.md:13]: nit: Does providing the following `toml` produce that JSON output. This section header is a little confusing.

**Review by ndneighbor — APPROVED** [review]
Two nits, I may wanna spend some time fleshing this out explaining how it works to interested users. This is a good start however.

File: src/pages/deploy/config-as-code.md (hunk 2 above)
Comment (coffee-cup) [reply src/pages/deploy/config-as-code.md:13]: Yup exactly. Both formats are the same

**Review by ndneighbor — APPROVED** [review]

**Pull request merged**